### PR TITLE
JBIDE-20184 Launching with Java 7 in jbdevstudio.ini results in module…

### PR DESCRIPTION
**Pull request by dazarov**
… errors in subsequent launches with Java 8

### CODE REVIEW

File: foundation/plugins/org.jboss.tools.foundation.checkup/src/org/jboss/tools/foundation/checkup/JVMProblemDetector.java [hunk 1]
```diff
@@ -39,6 +43,9 @@
 import org.osgi.framework.Constants;
 import org.osgi.service.prefs.BackingStoreException;
 
+import com.ibm.icu.text.DateFormat;
```
Comment (maxandersen) [inline foundation/plugins/org.jboss.tools.foundation.checkup/src/org/jboss/tools/foundation/checkup/JVMProblemDetector.java]: why is it using these from ibm.icu and not plain java versions ?

**Comment by maxandersen** [comment]
looks "ok" but please add tests for checking it is sane.

**Comment by alexeykazakov** [comment]
Rebased and pushed to master.

File: foundation/plugins/org.jboss.tools.foundation.checkup/src/org/jboss/tools/foundation/checkup/JVMProblemDetector.java [hunk 2]
```diff
@@ -517,4 +547,99 @@ public int hashCode() {
...
+	/**
+	 * for test purpose
+	 * @param date string in yyyy-MM-dd HH:mm:ss.SSS 
+	 * @throws ParseException
+	 */
+	public void setEclipseStartTime(String str) throws ParseException{
+		eclipseStartTime = DATE_FORMATTER.parse(str).getTime();
+	}
+	
+	private long getEclipseStartTime() {
```
Comment (maxandersen) [inline foundation/plugins/org.jboss.tools.foundation.checkup/src/org/jboss/tools/foundation/checkup/JVMProblemDetector.java]: if you made this method protected you could just subclass it in tests to get the time you want and would not need to add a test only setEclipseStartTime method.

**Pull request closed without merging**